akka-http-tests/src/test/java/akka/http/javadsl/server/values/HeadersTest.java

```diff
@@ -37,21 +33,24 @@ public void testValueByName() {
     }
     @Test
     public void testOptionalValueByName() {
-        TestRoute route = testRoute(completeWithValueToString(XTestHeader.optionalValue()));
+        TestRoute route = testRoute(optionalHeaderValueByName("X-Test-Header", value -> complete(value.toString())));
 
         route
             .run(HttpRequest.create())
             .assertStatusCode(200)
-            .assertEntity("None");
+            .assertEntity("Optional.empty");
 
         route
             .run(HttpRequest.create().addHeader(testHeaderInstance))
             .assertStatusCode(200)
-            .assertEntity("Some(abcdef-test)");
+            .assertEntity("Optional[abcdef-test]");
     }
+    
+    /*
+    FIXME A replacement of the "instance" RequestVal feature would probably be to allow easy creation of a custom header in Java DSL, and extract -> re-apply that instead.
```

Comment:
I'm removing the test instead. I initially thought that the test's `.instance()` here was referring to an "instance" of a custom modelled Java header class. Those don't exist though, and `instance` just referred to an instance of generic `HttpHeader` name and value.

In the Scala DSL, you can't extract a generic `HttpHeader` instance when providing a specific name, so you can't do that in Java anymore now either. It's also a bit nonsensical: if you provide an exact name for the header, you already know its name.